src/elisp/treemacs.el

```diff
@@ -240,6 +240,97 @@ An error message is displayed if the current buffer is not part of any project."
              (treemacs-goto-file-node path)
            (treemacs-add-project-to-workspace path name)))))))
 
+;;;###autoload
+(defun treemacs--bookmark-handler (record)
```

Comment:
What should I name it, `treemacs-bookmarks.el` or `treemacs-bookmark-integration.el`? `treemacs-bookmark.el` would be logical, but my extension is already called that :P

Reply:
Oh, and should I move the other `treemacs-bookmark` things there as well? Then it could be required as `(with-eval-after-load (require 'treemacs-bookmark))` to prevent a hard dependency on `bookmark.el`.

Reply:
> What should I name it

Let's go with bookmarks.

> Then it could be required as (with-eval-after-load 'bookmark (require 'treemacs-bookmark)) to prevent a hard dependency on bookmark.el.

Good idea, let's do that :+1: